x-pack/plugins/cloud_security_posture/public/plugin.tsx

```diff
@@ -42,6 +48,18 @@ export class CspPlugin
   }
 
   public start(core: CoreStart, plugins: CspClientPluginStartDeps): CspClientPluginStart {
+    plugins.fleet.registerExtension({
+      package: CLOUD_SECURITY_POSTURE_PACKAGE_NAME,
+      view: 'package-policy-create',
+      Component: LazyCspCreatePolicy,
+    });
+
+    plugins.fleet.registerExtension({
+      package: CLOUD_SECURITY_POSTURE_PACKAGE_NAME,
+      view: 'package-policy-edit',
+      Component: LazyCspEditPolicy,
```

Comment:
in relations to previous comments, it seems that all it take is passing the `disableDeploymentTypeSelector` to a unified component  in the `view: 'package-policy-edit'` option

Reply:
as mentioned, it's intentional to keep them separated as they are different contexts using similar UI components.